crates/sui-adapter/src/execution_engine.rs

```diff
@@ -369,7 +369,13 @@ fn advance_epoch<S: BackingPackageStore + ParentSync + ChildObjectResolver>(
     for (version, modules) in change_epoch.system_packages.into_iter() {
         let modules: Vec<_> = modules
             .into_iter()
-            .map(|m| CompiledModule::deserialize(&m).unwrap())
+            .map(|m| {
```

Comment:
I'm tempted to say that we shouldn't impose the protocol config version here, but instead rely on the `get_available_system_package_bytes` process (which is the other half of framework upgrades) to confirm for us that the bytecode format that we have built the framework bundled with the system at is supported by the max bytecode version in the proposed next protocol config. (cc @mystenmark)

Reply:
This should say:

> is supported by the max bytecode version in the current protocol config.

If that's what we're set on, but note that if we make a mistake in the set-up I mentioned above (for the two-phase protocol + framework upgrade) this will prevent the protocol upgrade that bumps the max bytecode format version from going through.